clients/src/test/java/org/apache/kafka/clients/producer/internals/TransactionManagerTest.java

```diff
@@ -542,15 +522,93 @@ public void testAllowAbortOnProduceFailure() throws InterruptedException {
         assertTrue(transactionManager.isReadyForTransaction());  // make sure we are ready for a transaction now.
     }
 
-
     @Test
-    public void shouldNotAddPartitionsToTransactionWhenTopicAuthorizationFailed() throws Exception {
-        verifyAddPartitionsFailsWithPartitionLevelError(Errors.TOPIC_AUTHORIZATION_FAILED);
+    public void testHandlingOfUnknownTopicPartitionError() throws InterruptedException {
```

Comment:
Could we break this into two cases: one for AddPartitions, and one for TxnOffsetCommit?